src/events/SDL_dragevents.c

```diff
@@ -0,0 +1,64 @@
+#include "../SDL_internal.h"
+
+#include "SDL_events.h"
+#include "SDL_events_c.h"
+#include "SDL_dragevents_c.h"
+
+#include "../video/SDL_sysvideo.h"  /* for SDL_Window internals. */
+
+const int
+SDL_SendDrag(SDL_Window *window, const SDL_EventType evtype, const char *data, int x, int y)
+{
+    static SDL_bool app_is_dragging = SDL_FALSE;
+    int posted = 0;
+
+    const SDL_bool need_begin = window ? !window->is_dragging : !app_is_dragging;
+    SDL_Event event;
+
+    if (need_begin) {
+        SDL_zero(event);
+        event.type = SDL_DRAGBEGIN;
+        event.drag.x = x;
+        event.drag.y = y;
+
+        if (window) {
+            event.drag.windowID = window->id;
+        }
+
+        posted = (SDL_PushEvent(&event) > 0);
+        if (!posted) {
+            return 0;
+        }
```

Comment:
This assignment seems unnecessary, and a bit confusing. I'd expect `posted` to be read from again later, but it isn't. So it would be simpler to just check the result directly I think:

```suggestion
        if (SDL_PushEvent(&event) == 0) {
            return 0;
        }
```